internal/tscode/builder.tmpl

```diff
@@ -14,7 +14,7 @@
   {{$tsFieldName := $field.TsBuilderFieldName -}}
 
   {{/* we magically expect a local variable input to be defined when this is used */ -}}
-  if (input.{{$tsFieldName}} !== undefined) {
+  if (input.{{$tsFieldName}} !== undefined || this.operation === WriteOperation.Delete) {
```

Comment:
this makes sense. my guess is it expects only the remove parts to apply for deletes?

Reply:
In looking at this file now I might have to add this check in other places too. Let me do some checking before we merge this.

Reply:
Ok, finally got back around to this and double checked everything. This should be good to merge. 👍